### PR TITLE
Feature EPL-1742: First Model of the TabWidget Component

### PR DESCRIPTION
**Pull request by Sankku**
First model of the Widget implemented inside the Sidebar when a row is selected in the table this components render the information inside that row.

image

### CODE REVIEW

File: packages/ComponentLibrary/src/components/Widgets/TabWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+const TabWidget: React.FC<TabWidgetProps> = ({
+  onSave,
+  onCancel,
+  editButtonLabel = '',
+  cancelButtonLabel = '',
+  saveButtonLabel = '',
+  noRecordText = '',
+}) => {
+  const { selectedRecord, setSelectedRecord } = useRecordContext();
```
Comment (LuuchoRocha) [inline packages/ComponentLibrary/src/components/Widgets/TabWidget.tsx:19]: Would be possible to pass this via props, so we can avoid using the context directly in the component library?

File: packages/ComponentLibrary/src/components/FormView/FormSection.tsx [hunk 2]
```diff
@@ -79,15 +65,17 @@ const FormSection: React.FC<FormSectionProps> = ({
...
               <FormFieldGroup
                 name={key}
                 field={field}
                 onChange={onInputChange}
+                readOnly={readOnly}
               />
-              {index < fields.length - -1 && (index + 1) % 2 !== 0 && (
-                <Box sx={styles.dottedLine} />
-              )}
+              {index < fields.length + 1 &&
```
Comment (tomirodeghiero) [inline packages/ComponentLibrary/src/components/FormView/FormSection.tsx:75]: I think that is better:

```
{index <= fields.length &&
  (index + 1) % dottedLineInterval !== 0 && (
    <Box sx={styles.dottedLine} />
)}
```

**Review by LuuchoRocha — DISMISSED** [review]

File: packages/MainUI/src/screens/Home/styles.ts [hunk 3]
```diff
@@ -11,9 +11,15 @@ const styles = {
...
+  container: {
+    marginTop: '0.25rem',
+    overflow: 'hidden',
+    display: 'flex',
+    flexGrow: '1',
+  },
   sidebarPaper: {
     right: 4,
-    width: '32%',
+    maxWidth: '32%',
```
Comment (tomirodeghiero) [inline packages/MainUI/src/screens/Home/styles.ts:22]: This is strange 🤔

File: packages/ComponentLibrary/src/components/FormView/FormField.tsx [hunk 4]
```diff
@@ -77,7 +67,7 @@ const FormField: React.FC<{
         return (
           <Select
             iconLeft={<SearchOutlined sx={{ width: 24, height: 24 }} />}
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/FormView/FormField.tsx:69]: view this icon imported from mui core

File: packages/ComponentLibrary/src/components/Widgets/TabWidget.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    if (!selectedRecord) return null;
+
+    return (
+      <FormView
+        key={selectedRecord.id.value}
+        data={selectedRecord}
+        onSave={handleSave}
+        onCancel={handleCancel}
+        readOnly={!isEditing}
+        gridItemProps={{ xs: 12, sm: 1, md: 12 }}
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/Widgets/TabWidget.tsx:59]: styles in constant

File: packages/ComponentLibrary/src/components/Widgets/TabWidget.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        dottedLineInterval={12}
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/Widgets/TabWidget.tsx:60]: number in constant

**Review by leandroallemandismf — APPROVED** [review]

**Review by tomirodeghiero — APPROVED** [review]

File: packages/ComponentLibrary/src/components/FormView/FormField.tsx [hunk 7]
```diff
@@ -7,36 +6,30 @@ import {
...
   </Box>
 );
 
-const FormField: React.FC<{
-  name: string;
-  field: FieldDefinition;
-  onChange: (name: string, value: FieldDefinition['value']) => void;
-}> = ({ name, field, onChange }) => {
+const FormField: React.FC<FormFieldProps> = ({ name, field, onChange }) => {
+  const [value, setValue] = useState(field.value);
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/FormView/FormField.tsx:26]: add type FieldDefinition

File: packages/ComponentLibrary/src/components/Widgets/TabWidget.tsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    setIsEditing(false);
+  }, [onSave, selectedRecord, setSelectedRecord]);
+
+  const handleCancel = useCallback(() => {
+    setIsEditing(false);
+    setSelectedRecord(selectedRecord);
+    onCancel();
+  }, [onCancel, selectedRecord, setSelectedRecord]);
+
+  const handleFormChange = useCallback(() => {}, []);
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/Widgets/TabWidget.tsx:47]: add TODO comment

File: packages/MainUI/src/screens/Home/index.tsx [hunk 9]
```diff
@@ -89,16 +93,36 @@ const Home: React.FC = () => {
     ],
   );
 
+  useEffect(() => {
+    if (selectedRecord) {
+      const newWidgets = createWidgets(selectedRecord, setSelectedRecord).map(
+        widget => {
+          if (widget.id === '1') {
```
Comment (leandroallemandismf) [inline packages/MainUI/src/screens/Home/index.tsx:100]: Is this okay?

**Review by leandroallemandismf — APPROVED** [review]

**Review by tomirodeghiero — APPROVED** [review]

**Pull request merged**